### PR TITLE
Add support for money managed chart in host reports

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/4606

⚠️ Require: https://github.com/opencollective/opencollective-api/pull/6786

total-money-managed

### CODE REVIEW

File: components/host-dashboard/reports-section/TotalMoneyManagedSection.js [hunk 1]
```diff
@@ -37,7 +51,102 @@ const TotalFundsLabel = styled(Container)`
...
+    border-radius: 10px;
+    & > span {
+      vertical-align: middle;
+    }
+  }
+
+  .apexcharts-legend-marker {
+    margin-right: 8px;
+  }
+`;
```
Comment (Betree) [inline components/host-dashboard/reports-section/TotalMoneyManagedSection.js:69]: If we're going to reuse the same styles for all charts, this should be exported to a common component. We can revisit later, but as a starter having this in `components/ChartWrapper.js` with a description like this could be a good start:
> Wraps a chart from ApexCharts to plug our custom styles to it

**Review by Betree — APPROVED** [review]
Small refactors suggested below but everything looks good

File: components/host-dashboard/reports-section/HostFeesSection.js [hunk 2]
```diff
@@ -204,7 +204,7 @@ const getSeriesFromData = (intl, timeSeries) => {
   ];
 };
 
-const getActiveYearsOptions = host => {
+export const getActiveYearsOptions = host => {
```
[the remark this answers is not in the document]
Comment (SudharakaP) [reply components/host-dashboard/reports-section/HostFeesSection.js]: Makes sense. Done. 👍🏽

File: components/host-dashboard/reports-section/TotalMoneyManagedSection.js (hunk 1 above)
Comment (SudharakaP) [reply components/host-dashboard/reports-section/TotalMoneyManagedSection.js]: Thanks, I've extracted this to a separate component.

**Comment by SudharakaP** [comment]
@Betree : I've re-based this so that the new alignments that you've done to the reports section is also incorporated.

**Comment by Betree** [comment]
@SudharakaP I just merged the centralization of filter, it probably requires another rebase

**Comment by SudharakaP** [comment]
> @SudharakaP I just merged the centralization of filters, it probably requires another rebase

@Betree : Thanks for the heads up, I've re-based this, but it seems that there's some problem with the filters even on staging and the `Host Fees` and `Platform Tips` sections always show zeros. I'll tackle these in separate PRs, not to make this too long.

File: components/edit-collective/Menu.js [hunk 3]
```diff
@@ -216,7 +216,7 @@ const sectionsDisplayConditions = {
   [EDIT_COLLECTIVE_SECTIONS.PENDING_ORDERS]: () => false,
   [EDIT_COLLECTIVE_SECTIONS.SENDING_MONEY]: () => false,
   [EDIT_COLLECTIVE_SECTIONS.HOST_TWO_FACTOR_AUTH]: () => false,
-  [EDIT_COLLECTIVE_SECTIONS.HOST_VIRTUAL_CARDS]: () => false,
+  [EDIT_COLLECTIVE_SECTIONS.HOST_VIRTUAL_CARDS]: () => true,
```
Comment (Betree) [inline components/edit-collective/Menu.js:219]: Seems out of scope

**Review by Betree — COMMENTED** [review]
@SudharakaP There are changes related to virtual cards in this PR, could you check?

File: components/edit-collective/Menu.js (hunk 3 above)
Comment (SudharakaP) [reply components/edit-collective/Menu.js]: Ops, this is a mistake. I was checking something on the virtual cards page and forgot to revert it. sorry about that.

**Comment by SudharakaP** [comment]
> @SudharakaP There are changes related to virtual cards in this PR, could you check?

Sorry, I was testing something related to virtual cards and forgot to revert those changes. I've removed those changes and now everything should be okay. 🙂

File: components/host-dashboard/reports-section/TotalMoneyManagedSection.js [hunk 4]
```diff
@@ -45,7 +59,85 @@ const getMoneyManagedChartAreas = (collectivesBalance, hostBalance, currency, is
...
+      name: intl.formatMessage({ defaultMessage: 'Total Managed Amount' }),
+      data: dataToSeries(totalMoneyManagedProgressNodes),
+    },
+  ];
+};
+
 const TotalMoneyManagedSection = ({ host, isLoading }) => {
+  const intl = useIntl();
+  const [showMoneyManagedChart, setShowMoneyManagedChart] = useState(false);
+  const yearsOptions = useMemo(() => getActiveYearsOptions(null), [null]);
```
Comment (Betree) [inline components/host-dashboard/reports-section/TotalMoneyManagedSection.js:128]: Why pass `null` rather than the `host` here?

File: components/host-dashboard/reports-section/TotalMoneyManagedSection.js [hunk 5]
```diff
@@ -45,7 +59,85 @@ const getMoneyManagedChartAreas = (collectivesBalance, hostBalance, currency, is
...
+  const chartOptions = useMemo(() => getChartOptions(intl, host?.currency), [host?.currency]);
+  const [selectedYear, setSelectedYear] = useState(() => new Date().getFullYear());
+  const variables = getQueryVariables(host?.slug, selectedYear);
+  const { loading, data, previousData } = useQuery(totalMoneyManagedQuery, {
+    variables,
+    context: API_V2_CONTEXT,
+    skip: !showMoneyManagedChart,
```
Comment (Betree) [inline components/host-dashboard/reports-section/TotalMoneyManagedSection.js:135]: :ice_cream: Not a requested change: splitting in a separate component would be a cleaner approach than using `skip` IMO. Let's not block the work on this feedback.

File: components/host-dashboard/reports-section/helpers.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export const getActiveYearsOptions = host => {
```
Comment (Betree) [inline components/host-dashboard/reports-section/helpers.js:1]: Nice :+1:

File: components/host-dashboard/reports-section/TotalMoneyManagedSection.js [hunk 7]
```diff
@@ -77,19 +169,63 @@ const TotalMoneyManagedSection = ({ host, isLoading }) => {
...
+            <FormattedMessage
+              id="Host.Metrics.TotalMoneyManages.description"
+              defaultMessage="Total amount held in your bank account for the Host and its Collectives."
+            />
+          </P>
+        </Container>
+        <Container px={2} textAlign="right">
+          <StyledLinkButton color="#46347F" asLink onClick={() => setShowMoneyManagedChart(!showMoneyManagedChart)}>
+            <P fontSize="12px" fontWeight="400" mt="16px">
+              <FormattedMessage defaultMessage="See historical" />
```
Comment (Betree) [inline components/host-dashboard/reports-section/TotalMoneyManagedSection.js:184]: From latest feedback about wording

```suggestion
              <FormattedMessage defaultMessage="See historic" />
```

File: components/host-dashboard/reports-section/TotalMoneyManagedSection.js [hunk 8]
```diff
@@ -77,19 +169,63 @@ const TotalMoneyManagedSection = ({ host, isLoading }) => {
...
+        <Container px={2}>
+          <P fontSize="12px" fontWeight="400" mt="16px">
+            <FormattedMessage
+              id="Host.Metrics.TotalMoneyManages.description"
+              defaultMessage="Total amount held in your bank account for the Host and its Collectives."
+            />
+          </P>
+        </Container>
+        <Container px={2} textAlign="right">
+          <StyledLinkButton color="#46347F" asLink onClick={() => setShowMoneyManagedChart(!showMoneyManagedChart)}>
```
Comment (Betree) [inline components/host-dashboard/reports-section/TotalMoneyManagedSection.js:182]: As mentioned in other PRs, we don't want purple on that page; we want to use the primary color of the host.

```suggestion
          <StyledLinkButton asLink onClick={() => setShowMoneyManagedChart(!showMoneyManagedChart)}>
```

File: components/host-dashboard/reports-section/TotalMoneyManagedSection.js (hunk 4 above)
Comment (SudharakaP) [reply components/host-dashboard/reports-section/TotalMoneyManagedSection.js]: Ops, corrected this one. 👍🏽

File: components/host-dashboard/reports-section/TotalMoneyManagedSection.js (hunk 5 above)
Comment (SudharakaP) [reply components/host-dashboard/reports-section/TotalMoneyManagedSection.js]: I've moved this out to a separate component. I think now it looks better. 🙂

**Review by Betree — APPROVED** [review]
Looking good!

**Comment by znarf** [comment]
Can you make sure this PR is updated from main?

**Comment by SudharakaP** [comment]
> Can you make sure this PR is updated from main?

Just updated both this and the API PR. 👍🏽

**Review by znarf — CHANGES_REQUESTED** [review]
Could you reverse the years in the dropdown, like in the Host Fees section. 2021 first, 2020 second, etc ...

**Review by znarf — CHANGES_REQUESTED** [review]
Could you look into number formatting, it's not readable right now. It's acceptable to merge this PR in this state but we need a follow up issue.

**Comment by SudharakaP** [comment]
> Could you reverse the years in the dropdown, like in the Host Fees section. 2021 first, 2020 second, etc ...

@znarf : Thanks. I've made the changes. 👍🏽

**Review by znarf — APPROVED** [review]
Good for me
